### ISSUE TITLE
Drush Runserver Not Accessible in Gitlab via Curl

### ISSUE THREAD

**Issue by mikemadison13** [mikemadison13/blt-gitlab-pipelines#8]
This is a similar issue to one in the past where the Bethat stuff wasn't running in CI because it was trying to use lando config from the local. Well, this is one where we need lando config.

```
> tests:server
> tests:frontend
> tests:server:start
[info] Killing all processing containing string 'runserver'...
[info] Killing all processes on port '8888'...
Launching PHP's internal web server via drush.
[info] Running server at http://127.0.0.1:8888...
[info] Waiting for non-50x response from http://127.0.0.1:8888...
[info] Waiting for non-50x response from http://127.0.0.1:8888...
[info] Waiting for non-50x response from http://127.0.0.1:8888...
[info] Skipped frontend-test target hook. No hook is defined.
> tests:server:kill
[info] Killing all processing containing string 'runserver'...
[info] Killing all processes on port '8888'...
> tests:behat
> tests:server:start
[info] Killing all processing containing string 'runserver'...
[info] Killing all processes on port '8888'...
Launching PHP's internal web server via drush.
[info] Running server at http://127.0.0.1:8888...
[info] Waiting for non-50x response from http://127.0.0.1:8888...
[info] Waiting for non-50x response from http://127.0.0.1:8888...
[Filesystem\FilesystemStack] mkdir ["/builds/mikemadison13/d9canary/reports"]
[info] Killing running google-chrome processes...
[info] Killing all processes on port '9222'...
[info] Launching headless chrome...
[Robo\Common\ProcessExecutor] Running 'google-chrome' --headless --disable-web-security --remote-debugging-port=9222  http://localhost in /builds/mikemadison13/d9canary
[info] Waiting for non-50x response from http://localhost:9222...
[info] Waiting for non-50x response from http://localhost:9222...
[info] Waiting for non-50x response from http://localhost:9222...
[info] Waiting for non-50x response from http://localhost:9222...
[info] Waiting for non-50x response from http://localhost:9222...
[info] Waiting for non-50x response from http://localhost:9222...
[info] Waiting for non-50x response from http://localhost:9222...
[info] Waiting for non-50x response from http://localhost:9222...
[info] Waiting for non-50x response from http://localhost:9222...
[info] Waiting for non-50x response from http://localhost:9222...
[info] Waiting for non-50x response from http://localhost:9222...
[info] Waiting for non-50x response from http://localhost:9222...
[info] Waiting for non-50x response from http://localhost:9222...
[info] Waiting for non-50x response from http://localhost:9222...
[info] Waiting for non-50x response from http://localhost:9222...
[info] Waiting for non-50x response from http://localhost:9222...
[info] Waiting for non-50x response from http://localhost:9222...
[info] Waiting for non-50x response from http://localhost:9222...
[info] Waiting for non-50x response from http://localhost:9222...
[info] Waiting for non-50x response from http://localhost:9222...
[info] Waiting for non-50x response from http://localhost:9222...
[info] Waiting for non-50x response from http://localhost:9222...
[info] Waiting for non-50x response from http://localhost:9222...
[info] Waiting for non-50x response from http://localhost:9222...
[info] Waiting for non-50x response from http://localhost:9222...
[info] Waiting for non-50x response from http://localhost:9222...
[info] Waiting for non-50x response from http://localhost:9222...
[info] Waiting for non-50x response from http://localhost:9222...
[info] Waiting for non-50x response from http://localhost:9222...
[info] Waiting for non-50x response from http://localhost:9222...
[info] Waiting for non-50x response from http://localhost:9222...
[info] Waiting for non-50x response from http://localhost:9222...
[info] Waiting for non-50x response from http://localhost:9222...
[info] Waiting for non-50x response from http://localhost:9222...
[info] Waiting for non-50x response from http://localhost:9222...
[info] Waiting for non-50x response from http://localhost:9222...
[info] Waiting for non-50x response from http://localhost:9222...
[info] Waiting for non-50x response from http://localhost:9222...
[info] Waiting for non-50x response from http://localhost:9222...
[info] Waiting for non-50x response from http://localhost:9222...
[info] Waiting for non-50x response from http://localhost:9222...
[info] Waiting for non-50x response from http://localhost:9222...
[info] Waiting for non-50x response from http://localhost:9222...
[info] Waiting for non-50x response from http://localhost:9222...
[info] Waiting for non-50x response from http://localhost:9222...
[info] Waiting for non-50x response from http://localhost:9222...
[info] Waiting for non-50x response from http://localhost:9222...
[info] Waiting for non-50x response from http://localhost:9222...
[info] Waiting for non-50x response from http://localhost:9222...
[info] Waiting for non-50x response from http://localhost:9222...
[info] Waiting for non-50x response from http://localhost:9222...
[info] Waiting for non-50x response from http://localhost:9222...
[info] Waiting for non-50x response from http://localhost:9222...
[info] Waiting for non-50x response from http://localhost:9222...
[info] Waiting for non-50x response from http://localhost:9222...
[info] Waiting for non-50x response from http://localhost:9222...
[info] Waiting for non-50x response from http://localhost:9222...
[info] Waiting for non-50x response from http://localhost:9222...
[info] Waiting for non-50x response from http://localhost:9222...
[info] Waiting for non-50x response from http://localhost:9222...
[info] Killing running google-chrome processes...
[info] Killing all processes on port '9222'...
> tests:server:kill
[info] Killing all processing containing string 'runserver'...
[info] Killing all processes on port '8888'...
[error]  Timed out.
For troubleshooting guidance and support, see https://docs.acquia.com/blt/support/ 

```

**Comment by mikemadison13** [comment]
Digging into this, it appears to be a commonly known issue in Gitlab because we are using a Docker image and you cannot curl against a service running inside Docker. see articles like:

- https://gitlab.com/gitlab-org/gitlab-selenium-server/-/issues/1
- https://gitlab.com/gitlab-org/gitlab-foss/-/issues/40774
- https://gitlab.com/gitlab-org/gitlab-runner/-/issues/4484

While this looks related to the BLT issue https://github.com/acquia/blt/issues/3968 it's only "presenting" in the same way... in that the wait / timeout detection is failing to identify that the service is actually running (because Gitlab is giving us back):

```
[error]  cURL error 7: Failed to connect to localhost port 8888: Connection refused (see https://curl.haxx.se/libcurl/c/libcurl-errors.html)
Drush logged the following errors while attempting to start the web server:
For troubleshooting guidance and support, see https://docs.acquia.com/blt/support/ 
```

I have confirmed that commenting out the executor -- https://github.com/acquia/blt/blob/12.x/src/Robo/Common/Executor.php#L156 --  wait check function results in Behat tests that just work. So the service is running and google-chrome CAN get to it but the curl check is failing. 

Other useful links:

- https://docs.gitlab.com/ee/ci/docker/using_docker_images.html#accessing-the-services

**Comment by mikemadison13** [comment]
Lindsey proposed some fixes similar to this but the 0.0.0 trick doesn't seem to work, see https://github.com/acquia/blt/pull/3525

**Comment by mikemadison13** [comment]
further prodding at this indicates that the drush runserver url (127.0.0.1:8888) is actually returning ok from the curl. it's the localhost:9222 url from the chrome driver that is throwing the error.
